src/charm.py

```diff
@@ -30,6 +31,60 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
 
+    def _application_config(self) -> dict | None:
+        """Decode the value of the charm configuration application-config.
+
+        Returns:
+            The value of the charm configuration application-config.
+
+        Raises:
+            ReconciliationError: when application-config is invalid.
+        """
+        try:
+            config = self.model.config["application-config"]
+            if not config:
+                return None
+            application_config = json.loads(config)
+            if isinstance(application_config, dict):
+                return application_config
+            raise ReconciliationError(
+                new_status=BlockedStatus(
+                    "Invalid application-config value, expecting an object in JSON"
+                )
+            )
+        except json.JSONDecodeError as exc:
+            raise ReconciliationError(
+                new_status=BlockedStatus("Invalid application-config value, expecting JSON")
```

Comment:
Should this be logged as well to show what was actually received? That can often help with debugging

Reply:
That's a great idea. Added, thanks!

Reply:
The json decode error exception is not human-readable, it will be something like:
```
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```